src/interfaces/ViewInterface/ViewInterfaceContainer.js

```diff
@@ -31,6 +38,35 @@ export class ViewInterfaceContainer extends Component {
     });
   };
 
+  showConfirmUnassign = () => this.setState({ showConfirmUnassign: true });
+
+  hideConfirmUnassign = () => this.setState({ showConfirmUnassign: false });
+
+  showConfirmDelete = () => this.setState({ showConfirmDelete: true });
+
+  hideConfirmDelete = () => this.setState({ showConfirmDelete: false });
+
+  onUnassign = () => {
+    const { match, mutator, resources, showMessage } = this.props;
+    const org = get(resources, 'organization.records.0');
+    const interfaceId = get(match, 'params.id');
+
+    this.hideConfirmUnassign();
+    unassignInterface(mutator.organization, interfaceId, org)
+      .then(() => this.onClose())
+      .catch(() => showMessage('ui-organizations.interface.message.unassigned.fail', 'error'));
+  };
+
+  onDelete = () => {
+    const { match, mutator, resources } = this.props;
+    const org = get(resources, 'organization.records.0');
+    const interfaceId = get(match, 'params.id');
+
+    this.hideConfirmDelete();
+    deleteInterface(mutator.organization, mutator.interface, interfaceId, org)
+      .finally(() => this.onClose());
```

Comment:
basically, it's a wrong logic. The page should be closed on `then`, when interface has been deleted without errors. on `catch` you have to show error notification. The way you've got from contacts was implemented due to bug in stripes-connect, but it has been fixed already.

Reply:
If you could manage to make things right here it would be great if you can make the same fix for contacts

Reply:
fixed

Reply:
thanks